### PR TITLE
Add missing possible start rules in python.gram

### PR DESCRIPTION
**Pull request by lysnikolaou**

### CODE REVIEW

File: Grammar/python.gram [hunk 1]
```diff
@@ -239,7 +246,14 @@ star_named_expression[expr_ty]:
 named_expression[expr_ty]:
     | a=NAME ':=' b=expression { _Py_NamedExpr(CHECK(set_expr_context(p, a, Store)), b, EXTRA) }
     | expression
-annotated_rhs[expr_ty]: yield_expr | expressions
+
+annotated_rhs[expr_ty]: yield_expr | star_expressions
+
+expressions[expr_ty]:
+    | a=expression b=(',' c=expression { c })+ [','] {
+        _Py_Tuple(CHECK(seq_insert_in_front(p, a, b)), Load, EXTRA) }
```
Comment (gvanrossum) [inline Grammar/python.gram:254]: Can this not be expressed with the `'.'` operator?

File: Lib/test/test_peg_parser.py [hunk 2]
```diff
@@ -631,6 +631,12 @@ def f():
     ),
 }
 
+EXPRESSIONS_TEST_CASES = [
+    ("expression_add", "1+1"),
+    ("expression_add_2", "a+b"),
+    ("expression_call", "f(a, b=2, **kw)"),
```
Comment (gvanrossum) [inline Lib/test/test_peg_parser.py:637]: I see nothing with commas -- is that okay?

File: Modules/peg_parser.c [hunk 3]
```diff
@@ -2,22 +2,33 @@
...
     }
 
+    int mode;
+    if (strcmp(mode_str, "exec") == 0) {
+        mode = Py_file_input;
+    } else if (strcmp(mode_str, "eval") == 0) {
+        mode = Py_eval_input;
+    } else {
+        return PyErr_Format(PyExc_ValueError, "mode must be either 'exec' or 'eval'");
+    }
```
Comment (gvanrossum) [inline Modules/peg_parser.c:22]: If you're parsing from a file, why is "eval" needed?

File: Modules/peg_parser.c [hunk 4]
```diff
@@ -29,22 +40,33 @@ _Py_parse_file(PyObject *self, PyObject *args)
...
```
Comment (gvanrossum) [inline Modules/peg_parser.c:60]: Shouldn't you also support mode "single", like the compile() builtin?

File: Include/compile.h [hunk 5]
```diff
@@ -102,4 +102,7 @@ PyAPI_FUNC(int) _PyAST_Optimize(struct _mod *, PyArena *arena, _PyASTOptimizeSta
 #define Py_eval_input 258
 #define Py_func_type_input 345
 
+/* This doesn't need to match anything */
+#define Py_fstring_input 0
```
Comment (gvanrossum) [inline Include/compile.h:106]: Well, I guess it specifically must match nothing else? And zero is an easy value to get by accident. Why not make it something way out of range?

File: Parser/pegen/pegen.c [hunk 6]
```diff
@@ -546,7 +546,7 @@ Parser_Free(Parser *p)
 }
 
 Parser *
-Parser_New(struct tok_state *tok, START_RULE start_rule_func, int input_mode,
+Parser_New(struct tok_state *tok, int start_rule_func, int input_mode,
```
Comment (gvanrossum) [inline Parser/pegen/pegen.c:549]: Hm, now `_func` is an odd suffix for what's clearly a multi-valued flag.

File: Lib/test/test_peg_parser.py (hunk 2 above)
Comment (lysnikolaou) [reply Lib/test/test_peg_parser.py:637]: Good point!

File: Grammar/python.gram (hunk 1 above)
Comment (lysnikolaou) [reply Grammar/python.gram:254]: I don't think so, since there is a `+` after the parentheses (not a '*').

File: Modules/peg_parser.c (hunk 3 above)
Comment (lysnikolaou) [reply Modules/peg_parser.c:22]: It's not really needed, but should we disallow this?

Comment (gvanrossum) [reply Modules/peg_parser.c:22]: I see no use.

Comment (gvanrossum) [reply Modules/peg_parser.c:22]: I see no use.

**Comment by lysnikolaou** [comment]
> LGTM. (We will have to add something for `func_type_input` later, but that can wait.)
> 
> PS. Could you create issues for tasks needed to get all tests to parse?

Of course. I'll do that.

**Comment by lysnikolaou** [comment]
There is one more thing I missed. `statement_newline` should also accept a lone `NEWLINE`, but I'm not sure what the action should be. I tried to test this, bit didn't find an answer. For example, `ast.parse('\n', mode='single')` doesn't work.

I'm guessing it should be an empty `asdl_seq *`, but I'm not sure.

**Comment by lysnikolaou** [comment]
Turns out, it just generates a `Pass` node. (https://github.com/we-like-parsers/cpython/blob/33b77af56e50fbc150b704ab48bbdd424f021fe7/Python/ast.c#L831)

**Pull request merged**